src/desktop/lib/global_client_setup.tsx

```diff
@@ -168,3 +170,44 @@ function mountStitchComponents() {
     wrapper: globalReactModules.StitchWrapper,
   })
 }
+
+function trackAuthenticationEvents() {
+  const modes = ["login", "signup"]
+
+  modes.forEach(mode => {
+    if (Cookies.get(`analytics-${mode}`)) {
+      const data = JSON.parse(Cookies.get(`analytics-${mode}`))
+      Cookies.expire(`analytics-${mode}`)
+      const user = sd && sd.CURRENT_USER
+
+      if (user) {
+        const options = {
+          ...data,
+          userId: user.id,
+        }
+
+        let analyticsOptions
+        switch (mode) {
+          case "login":
+            analyticsOptions = successfullyLoggedIn(options)
+            break
+          case "signup":
+            analyticsOptions = createdAccount(options)
+            break
+        }
+        window.analytics.track(analyticsOptions)
+        analyticsIdentify(user)
+      }
+    }
+  })
+}
```

Comment:
#minor we iterate over the modes array and it isn't super apparent that we should only ever really have one analytics mode set at a time (because the logic to set the cookie doesn't live in this file). It might be worth adding a comment about that and verifying with a test that if a cookie  is set for `analytics-login` ,  `createdAccount(options)` doesn't get called or something like that...?

Reply:
I've updated the comment to point back to where we've set the cookie, but I'm unsure which test case is missing, can you elaborate?
We test setting the cookie here: https://github.com/artsy/force/pull/5549/files#diff-b814803748b6ec603aba8f14394f6ebbR61
And ensure the correct analytics event is fired here: https://github.com/artsy/force/pull/5549/files#diff-b035f52e99721979e24023de3ce65b8fR12

Reply:
I think the comment is enough. I was just thinking a test could have also served documentation. It was more of an either...or. 
 
I also realize that I was looking at a previous implementation where you had the `createdAccount` and `successfullyLoggedIn` helpers. This implementation is a bit more clear. Sorry for the confusion.